decaymap/decaymap_test.go

```diff
@@ -29,3 +29,31 @@ func TestImpl(t *testing.T) {
 		t.Error("got value even though it was supposed to be expired")
 	}
 }
+
+func TestCleanup(t *testing.T) {
+	dm := New[string, string]()
+
+	dm.Set("test1", "hi1", 1*time.Second)
+	dm.Set("test2", "hi2", 2*time.Second)
+	dm.Set("test3", "hi3", 3*time.Second)
+
+	time.Sleep(2 * time.Second)
```

Comment:
Nitpick: [Time is not a synchronization primitive](https://xeiaso.net/blog/nosleep/), can you change this to use the [Impl.expire](https://github.com/TecharoHQ/anubis/blob/db5143ae7a53eb54fadeb2166252e1a7de5450b9/decaymap/decaymap.go#L33-L49) method? This makes things less flakey when it comes to handling time edge cases such as leap seconds (we are due for a negative leap second sometime soon), daylight savings, and leap years.

Reply:
Ah, thanks for the cool read! Pretty cool to be able to link to your own blog about it lol.